include/libp2p/transport/tcp/bytes_counter.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <atomic>
+
+namespace libp2p::transport {
+
+  class ByteCounter {
+   private:
+    std::atomic<uint64_t> bytes_read_{0};
+    std::atomic<uint64_t> bytes_written_{0};
+
+    ByteCounter() = default;
+
+   public:
```

Comment:
I was surprised that there is no method to reset the counters. Well, that is another possible approach.